scripts/setup_sgtm_tasks_project.py

```diff
@@ -0,0 +1,193 @@
+#!/usr/bin/env python3
+
+import argparse
+from dataclasses import dataclass
+from typing import List
+
+import asana
+
+
+@dataclass
+class EnumOption:
+    name: str
+    color: str
+
+
+@dataclass
+class CustomField:
+    name: str
+    enum_options: List[EnumOption]
+
+
+CUSTOM_FIELDS = [
```

Comment:
I just thought of something -- is it possible for customers to have access to the API but not custom fields? Is it safe to assume that anyone using SGTM is on a non-free tier? I'll follow-up with the API team about this.

Reply:
OK - just confirmed https://asana.slack.com/archives/C0PHY07V2/p1616773926047500 -- I think we need to cover the case where a domain doesn't have access to custom fields. SGTM will still work with projects that don't have the custom fields, so we should be able to detect this somehow and not attempt to add the custom fields. Happy to pair on ideas here

Reply:
Ah that's a good point - I didn't think of that. I followed up on the thread and Kevin mentioned that there isn't a way to get premiumness via the API so we might need to think of an alternate solution.

One thing we could do is try to create this irrespective of premiumness, and I assume that the API should return an error if the operation isn't possible. I've asked the API team if this is true

Reply:
Response:
> You'll get a 402 status code with Custom Fields are not available for free users or guests as the message

Reply:
Slightly concerned about the guest part here based on the instructions in our setup